File: clients/ruby/model/player_view.rb
require_relative 'game'
class PlayerView
    attr_accessor :my_id
    attr_accessor :game
    def initialize(my_id, game)
        @my_id = my_id
        @game = game
    end
    def self.read_from(stream)
        my_id = stream.read_int()
        game = Game.read_from(stream)
        PlayerView.new(my_id, game)
    end
    def write_to(stream)
        stream.write_int(@my_id)
        @game.write_to(stream)
    end
end

File: clients/python/model/player_view.py
from .game import Game
class PlayerView:
    def __init__(self, my_id, game):
        self.my_id = my_id
        self.game = game
    @staticmethod
    def read_from(stream):
        my_id = stream.read_int()
        game = Game.read_from(stream)
        return PlayerView(my_id, game)
    def write_to(self, stream):
        stream.write_int(self.my_id)
        self.game.write_to(stream)
    def __repr__(self):
        return "PlayerView(" + \
            repr(self.my_id) + "," + \
            repr(self.game) + \
            ")"
